test/test_evaluation/test_train_evaluator.py

```diff
@@ -745,6 +748,223 @@ def test_get_splitter(self, te_mock):
         cv = evaluator.get_splitter(D)
         self.assertIsInstance(cv, sklearn.model_selection._split.KFold)
 
+    @unittest.mock.patch.object(TrainEvaluator, "__init__")
+    def test_get_splitter_cv_object(self, te_mock):
```

Comment:
It would be great if you could add a dummy use of the iterator to check that it actually returns something? Apparently, scikit-learn only checks some of the arguments at call time, not at instantiation time, see for example:
```python
import sklearn.model_selection
import sklearn.datasets

X, y = sklearn.datasets.load_boston(return_X_y=True)
grouped = sklearn.model_selection.GroupKFold(n_splits=4)
for train_split, test_split in grouped.split(X, y):
    print(train_split, test_split)
```